contracts/src/v0.1/Prepayment.sol

```diff
@@ -150,14 +151,21 @@ contract Prepayment is Ownable, PrepaymentInterface, TypeAndVersionInterface {
         s_currentAccId++;
         uint64 currentAccId = s_currentAccId;
         address[] memory consumers = new address[](0);
-        s_accounts[currentAccId] = Account({balance: 0, reqCount: 0});
+        string memory accType = "regular";
```

Comment:
@KelvinThai Thank you for digging in to the problem! I also thought about this. I will try to explain it and let me know what you think.

We could create an extra mapping to store registered coordinators instead of looping through all coordinators which punishes `regular` accounts creators by increasing gas cost for them.

```Solidity
mapping(address => bool) private sIsCoordinator;
```

The values of `sIsCoordinator` will be updated inside of `addCoordinator` and `removeCoordinator`.

The `createAccount` function will then look similar to the one below.

```Solidity
function createAccount() external returns (uint64) {
    s_currentAccId++;
    uint64 currentAccId = s_currentAccId;
    address[] memory consumers = new address[](0);
    string memory accType = "reg";
    if (sIsCoordinator[msg.sender]) {
        accType = "tmp";
    }
    s_accounts[currentAccId] = Account({balance: 0, reqCount: 0, accType: accType});
    s_accountConfigs[currentAccId] = AccountConfig({
        owner: msg.sender,
        requestedOwner: address(0),
        consumers: consumers
    });

    emit AccountCreated(currentAccId, msg.sender, accType);
    return currentAccId;
}
```

Reply:
Totally agreed. A mapping would give faster and cheaper access to Coordinators. Let's move this way